internal/serverapi/serverapi.go

```diff
@@ -147,6 +147,31 @@ type CreateSnapshotSourceResponse struct {
 	SnapshotStarted bool `json:"snapshotted"` // whether snapshotting has been started
 }
 
+// DeletionResponseSingle contains response of attempted deletion of either snapshot or source.
+type DeletionResponseSingle struct {
+	Type    string `json:"type"`    // did we attempt to delete a snapshot or a source
+	Deleted bool   `json:"deleted"` // whether the object was deleted
+	Error   string `json:"error"`   // any errors
+}
+
+// DeletionResponseBatch contains response of attempted batch deletion.
+type DeletionResponseBatch struct {
+	Items []*DeletionStatus `json:"items"` // status of each item we attempted to delete
```

Comment:
do we anticipate this failing in any non-trivial ways (we don't really have locks on snapshots so this will either always succeed or always fail)? Thus I would say we may not need this response type either (at least for now)?